openpgp/read.go

```diff
@@ -461,3 +461,31 @@ func checkArmoredDetachedSignature(keyring KeyRing, signed, signature io.Reader)
 	}
 	return checkDetachedSignature(keyring, signed, body)
 }
+
+// CheckDesignatedRevokers will try to confirm any of designated
+// revocation of entity. For this function to work, revocation
+// issuer's key should be found in keyring, otherwise it will return
+// false, nil if keys are not found (meaning: revocation not
+// confirmed).
+func CheckDesignatedRevokers(keyring KeyRing, entity *Entity) (bool, *Key) {
```

Comment:
Still unsure about the API here. Entity can be basically in 3 states now, in regards to revocations:
1) Not revoked
2) Maybe revoked (if there are 3rd party revocations we couldn't verify)
3) Definitely revoked (if there are 1st or 3rd party revocations that were verified)